src/components/TodoApp.vue

```diff
@@ -24,26 +30,76 @@ import ToDoHeader from './ToDoHeader.vue'
 import EmptyListImage from './EmptyListImage.vue'
 import { Todo } from '../todo.ts'
 
-const todos = ref<Todo[]>(getFromLocalStorage())
+const incompleteTodos = ref<Todo[]>(getIncompletedFromLocalStorage())
+const completeTodos = ref<Todo[]>(getCompletedFromLocalStorage())
 const isShowingForm: Ref<boolean> = ref(false)
-const isShowingEmptyImage = computed(() => !isShowingForm.value && !todos.value)
-const i = -1
+const isShowingEmptyImage = computed(
+  () => !isShowingForm.value && !incompleteTodos.value.length && !completeTodos.value.length
+)
 const closeFormRef = ref(null)
+const i = -1
 const todo = ref<Todo>({
   title: 'Title',
   description: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.',
   priority: 'Medium',
   isChecked: false,
   dueDate: new Date(),
+  id: new Date().getTime(),
 })
 
 function saveToLocalStorage() {
-  localStorage.setItem('todos', JSON.stringify(todos.value))
+  localStorage.setItem('incompleteTodos', JSON.stringify(incompleteTodos.value))
+  localStorage.setItem('completeTodos', JSON.stringify(completeTodos.value))
+}
+
+function getIncompletedFromLocalStorage() {
+  const savedIncompleteTodos = localStorage.getItem('incompleteTodos')
+  return savedIncompleteTodos ? JSON.parse(savedIncompleteTodos) : []
+}
+
+function getCompletedFromLocalStorage() {
+  const savedCompleteTodos = localStorage.getItem('completeTodos')
+  return savedCompleteTodos ? JSON.parse(savedCompleteTodos) : []
+}
+
+function toggleCheck(checked: boolean, index: number) {
+  if (checked) {
+    const updatedTodo = incompleteTodos.value.find((todo) => todo.id === index)
+    if (updatedTodo) {
+      updatedTodo.isChecked = checked
+    }
+  } else {
+    const updatedTodo = completeTodos.value.find((todo) => todo.id === index)
+    if (updatedTodo) {
+      updatedTodo.isChecked = checked
+    }
+  }
+  if (checked) {
+    animateDown(index)
+  } else {
+    animateUp(index)
+  }
+}
+
+function animateDown(index: number) {
+  const copyToDo = incompleteTodos.value.find((todo) => todo.id === index)
```

Comment:
Actually you can use findIndex instead of find, so when removing the element you can use splice instead of filter.